src/kern/npf_inet.c

```diff
@@ -623,6 +624,11 @@ npf_cache_all(npf_cache_t *npc)
 		    sizeof(struct icmp6_hdr));
 		l4flags = NPC_LAYER4 | NPC_ICMP;
 		break;
+	case IPPROTO_GRE:
+		/* Cache: layer 4 - GRE. */
+		npc->npc_l4.hdr = nbuf_advance(nbuf, hlen, sizeof(pptp_gre_hdr_t));
```

Comment:
You can move this into a function (declare it in npf_impl.h for now) and thus eliminate the need for `npf_pptp_gre.h` header. I need to think what's the best way to handle this.  Maybe we should have a generic GRE handler (not specific to the PPTP ALG).  In any case, I can refactor/cleanup this later..